File: spring-rest-configuration-autodiscover.scala/src/main/scala/demo/service/example/ExamplePropertyBooleanService.scala
package demo.service.example

import demo.service.core.{Code, PropertyBooleanService}
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Service

@Service
class ExamplePropertyBooleanService extends PropertyBooleanService {

  @Value("${example.property.boolean}")
  private val defaultValue: Boolean = null

  override def getCode: Code.Value = Code.EXAMPLE_BOOLEAN_PROPERTY

  override def getDefaultValue: Boolean = defaultValue
}

File: spring-rest-configuration-autodiscover.groovy/src/main/groovy/demo/service/example/ExamplePropertyBooleanService.kt
package demo.service.example

import demo.service.core.Code
import demo.service.core.PropertyBooleanService
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Service

@Service
class ExamplePropertyBooleanService : PropertyBooleanService() {

    @Value("\${example.property.boolean}")
    private val defaultValue: Boolean? = null

    override fun getCode() = Code.EXAMPLE_BOOLEAN_PROPERTY

    public override fun getDefaultValue() = defaultValue!!
}
